packages/xstate-react/test/useMachine.test.tsx

```diff
@@ -647,4 +647,57 @@ describe('useMachine (strict mode)', () => {
     );
     done();
   });
+
+  // https://github.com/davidkpiano/xstate/issues/1334
+  it('should work with a rehydrated state', (done) => {
```

Comment:
there are already 2 tests here which are supposed to handle rehydrated state:
https://github.com/davidkpiano/xstate/blob/9ecefca84bfc3db6df09c41b884ac00ef0f6efb1/packages/xstate-react/test/useMachine.test.tsx#L100
https://github.com/davidkpiano/xstate/blob/9ecefca84bfc3db6df09c41b884ac00ef0f6efb1/packages/xstate-react/test/useMachine.test.tsx#L113

How this one is different? I understand that this relates to delayed transitions somehow, but its not obvious for me why this has been failing before - what info was lacking (and where) that this has failed? I would understand more if this would be about rehydrating `doingStuff` state but this is, from what I understand, rehydrating `idle` and I don't see how this affects a delayed transition in another state.

I know that this fix is OK - it has fixed the issue after all, but would love to understand the core of the issue better.

Reply:
Nothing in the state was lacking - all the available information is available as expected.

The issue was that the machine never had a chance to "initialize" (resolve) its delayed transitions because `machine.initialState` was never read (since we were starting directly from the resolved state).

Ideally, initialization happens in the constructor, but that would currently break "target getters" (which are deprecated in v5).

Reply:
**Before:**

- Machine defined, delayed transitions not resolved
- Interpreter starts at restored state
- ❌ Delayed transitions still do not exist; will not be taken

**After:**

- Machine defined, delayed transitions not resolved
- `machine.initialState` read which causes the machine to initialize its transitions (resolve its delayed transitions, etc.)
  - I know this is a code smell; will be fixed in v5
- Interpreter starts at restored state
- ✅ Delayed transitions exist; will be taken

Reply:
Ok - I think I get it now. Thanks for the explanation. 👍